File: WdtcConsole/src/main/java/org/wdt/wdtc/console/WdtcMain.java
package org.wdt.wdtc.console;

import org.apache.commons.cli.Options;
import org.apache.log4j.Logger;
import org.wdt.wdtc.console.manger.CommandTaskManger;
import org.wdt.wdtc.core.auth.yggdrasil.AuthlibInjector;
import org.wdt.wdtc.core.manger.GameFileManger;
import org.wdt.wdtc.core.manger.TaskManger;
import org.wdt.wdtc.core.utils.WdtcLogger;

import java.util.Arrays;

public class WdtcMain {

  private static final Logger logmaker = WdtcLogger.getLogger(WdtcMain.class);

  static {
    TaskManger.ckeckVMConfig();
  }

  public static void main(String[] args) {
    try {
      TaskManger.runStartUpTask();
      AuthlibInjector.updateAuthlibInjector();
      GameFileManger.downloadVersionManifestJsonFileTask();
      Options options = new Options();
      logmaker.info("Args: " + Arrays.toString(args));
      CommandTaskManger commandTaskManger = new CommandTaskManger(options, args);
      commandTaskManger.startTask();
    } catch (Throwable e) {
      logmaker.error(WdtcLogger.getExceptionMessage(e));
      throw new RuntimeException(e);
    }
  }
}

File: WdtcConsole/src/main/kotlin/org/wdt/wdtc/console/WdtcMain.kt
package org.wdt.wdtc.console

import org.apache.commons.cli.Options
import org.wdt.wdtc.console.manger.CommandTaskManger
import org.wdt.wdtc.core.auth.yggdrasil.updateAuthlibInjector
import org.wdt.wdtc.core.manger.ckeckVMConfig
import org.wdt.wdtc.core.manger.downloadVersionManifestJsonFileTask
import org.wdt.wdtc.core.manger.runStartUpTask
import org.wdt.wdtc.core.utils.getExceptionMessage
import org.wdt.wdtc.core.utils.logmaker

fun main(args: Array<String>) {
  try {
    ckeckVMConfig()

    runStartUpTask()
    updateAuthlibInjector()
    downloadVersionManifestJsonFileTask()

    val options = Options()
    logmaker.info("Args: " + args.contentToString())
    val commandTaskManger = CommandTaskManger(options, args)
    commandTaskManger.startTask()
  } catch (e: Throwable) {
    logmaker.error(e.getExceptionMessage())
    throw RuntimeException(e)
  }
}

